src/source/PeerConnection/PeerConnection.c

```diff
@@ -771,7 +780,27 @@ STATUS setRemoteDescription(PRtcPeerConnection pPeerConnection, PRtcSessionDescr
     CHK(remoteIceUfrag != NULL && remoteIcePwd != NULL, STATUS_SESSION_DESCRIPTION_MISSING_ICE_VALUES);
     CHK(pKvsPeerConnection->remoteCertificateFingerprint[0] != '\0', STATUS_SESSION_DESCRIPTION_MISSING_CERTIFICATE_FINGERPRINT);
 
-    CHK_STATUS(iceAgentStartAgent(pKvsPeerConnection->pIceAgent, remoteIceUfrag, remoteIcePwd, pKvsPeerConnection->isOffer));
+    if (!IS_EMPTY_STRING(pKvsPeerConnection->remoteIceUfrag) &&
+        !IS_EMPTY_STRING(pKvsPeerConnection->remoteIcePwd) &&
+        STRNCMP(pKvsPeerConnection->remoteIceUfrag, remoteIceUfrag, LOCAL_ICE_UFRAG_LEN) != 0 &&
+        STRNCMP(pKvsPeerConnection->remoteIcePwd, remoteIcePwd, LOCAL_ICE_PWD_LEN) != 0) {
+
+        CHK_STATUS(generateJSONSafeString(pKvsPeerConnection->localIceUfrag, LOCAL_ICE_UFRAG_LEN));
+        CHK_STATUS(generateJSONSafeString(pKvsPeerConnection->localIcePwd, LOCAL_ICE_PWD_LEN));
+        CHK_STATUS(iceAgentRestart(pKvsPeerConnection->pIceAgent,
```

Comment:
I might be missing a link, but, is it necessary to pass `localIceUfrag` and `localIcePwd` as part of the function arguments to copy into pIceAgent? Why cannot we do this at the before the function call itself? Because, I do not see any allocations for `pIceAgent` happening inside this function anyways.

Reply:
we are passing in the local uFrag and uPwd when we do createIceAgent, iceAgentRestart is being consistent with that.